src/pages/ReportChangeWorkspacePage.tsx

```diff
@@ -45,9 +45,13 @@ function ReportChangeWorkspacePage({report}: ReportChangeWorkspacePageProps) {
                 return;
             }
             Navigation.goBack(ROUTES.REPORT_WITH_ID.getRoute(reportID));
-            changeReportPolicy(reportID, policyID);
+            if (isIOUReport(reportID) && isPolicyMember(currentUserLogin, policyID)) {
+                moveIOUReportToPolicy(reportID, policyID);
+            } else {
+                changeReportPolicy(reportID, policyID);
+            }
         },
-        [reportID],
+        [currentUserLogin, reportID],
```

Comment:
@waterim not your code, but we need to change the conditional on l66 to check for IOU or Expense report, otherwise opening the change-workspace page leads to a Not Found page

```
if (!isMoneyRequestReport(report) || isMoneyRequestReportPendingDeletion(report) || (!report.total && !report.unheldTotal)) {
```